### PR TITLE
Implement `@CommandDefault`

### LINKED ISSUE #6: Add @DefaultCommand(string description)
_Discussion found at https://github.com/BradleyChatha/jcli/pull/5#discussion_r516807169_

* Using `null` is weird and unintuitive, even if it's just a "learn once and get over it" kind of thing.

* Is it worth adding a UDA for this, when only a single default command will appear in most projects? Especially since the command itself can easily be named "DefaultCommand"?

I am leaning towards implementing it, since clarity is definitely something I'd like to have here, but again it's only for a single part of any tool's code that has this weird behaviour.

The default command does have special behaviour and limitations within `CommandLineInterface` though, so it also does make sense to make that difference more explicit beyond "passing null to a UDA".

If I can't think of a good reason *not* to do this by tomorrow, then I'll go ahead and implement it.

### PR DESCRIPTION
**Pull request by BradleyChatha**
TLDR; `@Command(null, "desc")` -> `@CommandDefault("desc")`

Old syntax now fails an assert with a user-friendly message:

```
Null command names are deprecated, please use @CommandDefault instead.
```

Closes #6

### CODE REVIEW

**Comment by BradleyChatha** [comment]
Don't think there's much holding this back, just wanted to open a PR just in case.

**Comment by andrey-zherikov** [comment]
Is it checked somewhere that `@CommandDefault` is used not more than once?

**Comment by BradleyChatha** [comment]
It's done as a runtime check https://github.com/BradleyChatha/jcli/blob/2e5536eeb15f5313b079b8c96971b3f909b4da4d/source/jaster/cli/core.d#L505 mostly because it was just easy to slap it there. For such a minor, (hopefully) uncommon error that's easy to fix and will be flagged up immediately via an exception, it might be a waste of time to move it into a compile-time check.

**Comment by BradleyChatha** [comment]
I realised *just* as I pushed that I forgot to switch branches for github action change. Meh.
